osu.Server.Queues.ScoreStatisticsProcessor/Processors/UserTotalPerformanceProcessor.cs

```diff
@@ -107,6 +109,10 @@ public async Task UpdateUserStatsAsync(UserStats userStats, int rulesetId, MySql
                 if (s.pp == null)
                     return true;
 
+                // Score must be a pass (safeguard - should be redundant with preserve flag).
```

Comment:
Is this assuming that all fail scores will be `preserve==0`?

Reply:
That is the assumption, yes. But thinking about it it may be possible to pin those scores in the future/now?

Maybe this should’ve been a filter in the query…

Reply:
nah pinning failed scores likely won't be allowed. they wouldn't have replays and feels pretty pointless